### PR TITLE
SOL-80 [Integrate] Update FE to include image when viewing, creating,…

### PR DESCRIPTION
**Pull request by jasonchuasun**
… and updating campaign

## Issue Link
- Backlog [Link](https://framgiaph.backlog.com/board/SPH_SOLIDITY?issueType=715986&assignee=375950)
- Devserver [Link](https://sph-solidity-5rgtsmzbo-johnpaulbaneravercel-sun-asterisk.vercel.app/)

## Target Page | Component | Function
- home page - `/`
- campaigns page - `/campaigns`
- profile page > campaigns tab - `/profile`
- create-campaign page - `/create-campaign`
- edit-campaign page - `/edit-campaign/[id]`

## Pre-condition
- n/a

## Steps | Commands
1. create a new campaign - enter a valid URL for an image in the form
2. view the campaign in the home page and campaigns page
3. edit the campaign that you own 

## Expected Output
- the set image (from the image link) should be reflected on the cards 
- should the image be invalid, display image placeholder instead
- editing a campaign should reflect the correct image URL and should be updateable to a new image link

## Definition of Done
- creating, editing, viewing campaign functionality is working as intended
- imageURL can be added to the newly created campaign

## Test View Points
- [ ] Add here

## Notes
- spacing of elements/cards is updated from grid to flex
- cards should wrap based on screen width

## Screenshots | Recordings
image
image
image
image

### General
- [ ] Does the code meet the requirements of the project?
- [ ] Is the code readable and maintainable?
- [ ] Are there any unnecessary or redundant lines of code?
- [ ] Are there any long methods or functions that should be refactored?
- [ ] Are there any performance issues?

### Functionality
- [ ] Are there any logical errors?
- [ ] Are edge cases handled properly?
- [ ] Are there any potential security vulnerabilities?

### Code Style
- [ ] Does the code adhere to the project's coding standards?
- [ ] Are variable and function names descriptive and meaningful?
- [ ] Is there proper use of comments and documentation?
- [ ] Is the code properly formatted? Are indentation, spacing, and formatting consistent?

### Security
- [ ] Is the website secure from common attacks?
- [ ] Are inputs properly validated and sanitized?
- [ ] Are sensitive data and credentials properly protected?

### Testing
- [ ] Are there sufficient test cases?
- [ ] Do the test cases cover all functionality?
- [ ] Are the test cases properly written and documented?

### Additional Comments
- [ ] Are there any other issues or concerns with the code?
- [ ] Are there any suggestions for improvement?
- [ ] Are there any positive aspects of the code that should be highlighted?

### CODE REVIEW

**Comment by ipilikent** [comment]
## Approved PR
- Can input URL image
- Can't input Videos or Files Link
- The validation messages are working

## Proof
<details>

Sol- 80 Integrate

</details>

Test Case: https://docs.google.com/spreadsheets/d/114GrEn-LQQ8saIvjz9zqenuk05qmO4dN/edit#gid=637322504
-----
- Total Test time: 1 hr.
- Total Comments: 0
- Total Iterations: 0

**Review by ipilikent — APPROVED** [review]

File: fe/src/pages/edit-campaign/[id].vue [hunk 1]
```diff
@@ -140,12 +152,14 @@ const getCampaign = async (): Promise<void> => {
         values.fullname = result[2];
         values.campaign = result[3];
         values.story = result[4];
-        values.goal = +ethers.formatEther(result[5]);
+        values.imageUrl = result[5];
+        values.goal = +ethers.formatEther(result[6]);
         values.date = getDateYMD(result[7]);
```
Comment (wylenjoan) [inline fe/src/pages/edit-campaign/[id].vue]:
```suggestion
        values.date = getDateYMD(result[8]);
```

File: fe/src/components/user/campaigns.vue [hunk 2]
```diff
@@ -58,9 +63,9 @@ const getUserCampaings = async (pageNumber: number) => {
           userCampaigns.value = result[0].map((campaign: BigNumberish[]) => {
             return {
               id: campaign[0],
-              imgSrc,
               title: campaign[3],
               description: campaign[4],
+              imageUrl: campaign[5],
               ethValue: ethers.formatEther(campaign[6]),
               daysLeft: getDaysLeft(campaign[7]),
```
Comment (wylenjoan) [inline fe/src/components/user/campaigns.vue]:
```suggestion
              daysLeft: getDaysLeft(campaign[8]),
```

File: fe/src/pages/edit-campaign/[id].vue [hunk 3]
```diff
@@ -140,12 +152,14 @@ const getCampaign = async (): Promise<void> => {
         values.fullname = result[2];
         values.campaign = result[3];
         values.story = result[4];
-        values.goal = +ethers.formatEther(result[5]);
-        values.date = getDateYMD(result[7]);
+        values.imageUrl = result[5];
+        values.goal = +ethers.formatEther(result[7]);
```
Comment (wylenjoan) [inline fe/src/pages/edit-campaign/[id].vue]:
```suggestion
        values.goal = +ethers.formatEther(result[6]);
```

**Review by wylenjoan — DISMISSED** [review]
Approved :white_check_mark:
Review time: 1hr
Comments: 3
Iterations: 2
Working:
- Included image in viewing, creating, and updating campaign.

**Review by wylenjoan — APPROVED** [review]

**Review by johnpaul-sun — APPROVED** [review]
Approved ✅

Review Time: 15 mins
Comments: 0
IT: 0

**Pull request merged**